### PR TITLE
Allow slices on ClosureExpression's params

### PR DESCRIPTION
**Pull request by SeaDve**
Follow up for https://github.com/gtk-rs/gtk4-rs/pull/703

### CODE REVIEW

File: gtk4/src/closure_expression.rs [hunk 1]
```diff
@@ -23,18 +23,22 @@ define_expression!(
 
 impl ClosureExpression {
     #[doc(alias = "gtk_closure_expression_new")]
-    pub fn new<F, R>(params: impl IntoIterator<Item = Expression>, callback: F) -> Self
+    pub fn new<F, R, I>(params: impl IntoIterator<Item = I>, callback: F) -> Self
     where
         F: Fn(&[Value]) -> R + 'static,
         R: ValueType,
+        I: AsRef<Expression> + Clone,
```
Comment (sdroege) [inline gtk4/src/closure_expression.rs:30]: Why not `impl IntoIterator<Item = impl AsRef<Expression>>`?

Comment (SeaDve) [reply gtk4/src/closure_expression.rs:30]: I think it's a bit more readable, but maybe just me

Comment (sdroege) [reply gtk4/src/closure_expression.rs:30]: Something for @bilelmoussaoui to decide then. Looks good to me apart from this :)

Comment (sdroege) [reply gtk4/src/closure_expression.rs:30]: Why does it need `Clone` though? `&Expression` is `Clone`

Comment (SeaDve) [reply gtk4/src/closure_expression.rs:30]: The compiler used to complain about it a while ago, but somehow it's now okay

**Comment by bilelmoussaoui** [comment]
Needs a cargo fmt otherwise looks good now

File: gtk4/src/closure_expression.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk4/src/closure_expression.rs]: > Why not `impl IntoIterator<Item = impl AsRef<Expression>>`

I'm also in favour of using something like this btw

**Review by sdroege — APPROVED** [review]

**Pull request merged**